### ISSUE TITLE
trading statistics library (database)

### ISSUE THREAD

**Issue by thainano** [binary-com/binary-bot#1260]
when you are trading, you just say something like:
noone will ever make profit in binary
or this strategy earns 10$ per day
without any statistics
you can easily say something is a scam or a holy grail with out any evidence right?
it is like a blind guy trying to cross road. that is why 70% traders lose money
i am planning to make something like, for example:
in "VOX100" if stochastic:  "K cross D up"/"BB squeeze strategy with ... condition..." and your bot purchase "RISE" for duration "1M"
=> you have "65%" win probability base on "1 year" sample by group ABC study
all in the brackets " " is data that a computer will autotrade to collect
so i would like any idea/ help because my major is business not finance/IT
i can do it by myself but very slow. and i think many of you are needing it very urgent to protect ur wallet
we can use this resource together for all the contributors
OR A MORE SIMPLE WAY is to find how to extract all tradingview/binary.com data (including indicators' value) down to the 1 tick timeframe (or another source if they do not have)

**Comment by binary-ex-machina** [comment]
@thainano i can say that 90% of the traders will lose money. 

You can extract all the binary.com tick data from any market (forex markets have a limited amount of years that you can check) since it's inception.

I can test any strategy that you want and a can say that the longer you use the same strategy, more it will go for 50% win rate. No matter the strategy or market, all of them falls in the Law of the Great Numbers. No strategy will be profitable (unless you have a good amount of money and are willing to make very little per day) if you let it running 24/7 on binary options.

**Comment by binarydream1** [comment]
@binary-ex-machina how much money do we need to become 10% of the trader (winner) ?

**Comment by binary-ex-machina** [comment]
@binarydream1 what i do to make money with binary options is to avoid (the max i can) emotions when i'm trading. It's the key in my daily trading, stay totally focused and not let my emotions or any external interference affect my trading ( i wake up at 5 AM, so it's very quiet and nobody is around me)

Focus and emotionless, then you need to understand how it works. Deeply. Very deeply. All the options, all the markets you can trade them and the rules. How much payout you get, how much duration you can use, what tools you can use in your favor (charts, indicators and so on)

The last thing is to have money to use for it. Money that you can lose. It's the most important part of the trading (any type) is to only invest money in what you can lose it. This way you are not affected by emotions.

Emotionless, Focus, Deep Knowledge, Money to lose. With all this aligned and tons of hours spent looking charts, testing indicators, doing all kinds of math, reading forums, chatting on groups about binary options, watching videos i can say I understand how the market behaves and I understand that is impossible to predict. 

I'm not talking about bots here, because this is the last part of the journey for me. Bots on binary options will never work running 24/7. None of them will be consistent on long term, because of the nature of binary options, it's like a casino game. You play for too long, you lose.

If you have the knowledge to create a very good manual strategy that you can make money on it, why would need a bot, right? I like this idea but i think with bots you can expand to places that you can't do alone a fast enough. 

can you make money with digits? yes you can and there is a lot of people making money with it. It's consistent? no. It's possible to use the same strategy and always have the same result? no. 

So what's the purpose of the bots? Imagine how fast you will need to be to place a 5 tick duration trade manually, recalculate the stake on loss, analyse the indicators and make the decision? A bot can do that for you.

If you plan to stay manually i advise durations above 1 hour and chart analysis with 1 hour or more too. Daily is the best and i trade a lot on forex with this time frame.

Binary options is a game and you need to understand how it works. I advise anyone that wants to be a binary options trader to study probability, to understand all the trade types and to try, try, try for several months on demo before going to real account.

**Comment by thainano** [comment]
@binary-ex-machina do you know what kind of strategy this guy is using?
[https://www.youtube.com/channel/UCznvbF6QLxudY-u2-eS2f5g](url)
it seems that he can make consistent profit from BO with shorter time frame

**Comment by amebamcare** [comment]
> @binary-ex-machina do you know what kind of strategy this guy is using?
> [https://www.youtube.com/channel/UCznvbF6QLxudY-u2-eS2f5g](url)
> it seems that he can make consistent profit from BO with shorter time frame.
> he did not explain anything (he does said some small talk in Vietnamese) but good trader can spot the strategy (which i am not)

@thainano  that is all along the fake facts about getting rich in 5 minutes and some funny youtube videos  saying like you can win  $20000/ in 20 minutes or even some one post videos like " your $10 becomes $10k in 10 minutes " like that and so on.... All are the people who want to attract high viewership on YT or other social media sites to avail  the click rate income for their content. If you watch these people's video tutorial nobody will tell or teach with their real live account. Almost all will be their virtual demo account. After that what they will do.? They will send a free copy of their so called "$20000/ in 20 minutes" bots and if you check it and use in real account you will be shocked,seeing that the real thing is different. Then you will ask the support of that BOT provider. That man will tell about the another bot which will be fetching  the same magnetic marketing words.." I have another for bot $20000/ in 20 minutes. But it needs a cost of bot priced at around $100 or 150$ "  Then in few days you may buy that second bot at the price $100 or 150$ and second time you will become over shocked ( or mad this time ) and realised lately that you have been cheated by the above bot provider. You keep on contacting the above said bot provider . but this time you wont get any reply or feedback from that cheater. { Probably that cheater will try the same old trick and tactic with another new innocent trader like you and keep on doing this ... thereby he may get around daily around $1000-1500 [ if it is minimum for 10 victims per day you calculate the amount=10x150)  in very short period without doing anything but sending such fake trade bots.  This is the long story about such fake videos tutorials about binary trading.
But, you may ask :-  is there any succeeded traders in BO trade.? Yes **one out of 10 million traders** can be successful trader even they be now multi millionaire. But they will not tell their secret in public forum or some times they may sell his method at unaffordable price tag.
So the better method for successful BO trade is do your own research and find a super mathematician who specialises in **DATA SCIENCE** and also a highly skilled higher degree software professional for his assistance . Hire these people on short term job basis  for doing design and development of a highly innovative Artificial Intelligent based Bot which shall be deployed on a top rated GPU cloud servers like Amazon or google Deep Learning AMI then start trade with that **Neural Network triggered** Binary tradebot and this time with very super strategy which already you might have installed by your hired Professionals on that bot Then reap the attractive profit with passive income generation method regularly .  This is the only practical way. All other methods are just waste of time..

**Comment by thainano** [comment]
@amebamcare can you share more about Neural Network triggered Binary tradebot
there is still very little information about it on the internet

**Comment by amebamcare** [comment]
@thainano  the subject knowledge on Neural Network Artificial Intelligence [A.I.] ,we cannot  avail just googling the key words. This is one of the most Advanced study  in present days. If you want to just gather knowledge on this subject, there are short term online study centres. But that is not suffice for trading here. If you are financially good, hire some **DATA SCIENTISTS with higher degree mathematics** and a High end skilled software Engineer to find a solution for development of real innovative A.I based Binary tradebot  with unbeatable strategy to deploy it on a ML cloud server I think your ultimate aim is focussed for development of such A.I. bots only. Then you hire such people above said and do the job at your taste. or other wise for Knowledge purpose there are academic Institution online teaching  the subject   just key in the Key words with google **Neural network, Deep learning, Machine Learning, Keras project, Tensor flow, Data science and applications, RNN, CNN, LSTM**,and so on.. But you must have some basic computer science theoretical knowledge to understand the subject. If not, go with the first option [Hiring professionals] Good luck.

**Comment by binary-ex-machina** [comment]
@amebamcare have you ever find someone to work with that? I imagine the cost to have such a bot running on an external server,  but my biggest problem will be the data not realistic because of the lagging and missing ticks. I already did some works with Tensor Flow and Machine learning working with charts and in all my tries, the lag and missing ticks made huge impact, since the system will create patterns and try to predict the next tick. Some times ticks don't come at all, so you will have this dark requests, with 0 response. Even if the machine learn and adapt to this, unless you pay an expert to create this in a few weeks, it will take a lot of time to create such a system.

I think it's better to invest time studying trading in general.

**Comment by amebamcare** [comment]
@binary-ex-machina  Yes definitely. The system using Neural Network would be a very deep subject to be implemented. But I know few people succeeded and even now they are multi millionaire in just Binary Trading itself. But what is the unrevealed part is they( the brilliant traders _ just not yet like to show their secret proprietary  trade system in the public media due to their privacy and also the fearness if they revealed such Bot system then everybody will do this trade and thereby  his winning probability may become shrink and profit become poor if everybody trade using advanced AI based bot. 
 But one thing I am very clear, sure  and guaranteed which 100% possible way  is
   
# **_The trade bots using A.I based is the only way for real profit making way, It is the proven one which is still implemented and run by very rare people [ one out of million traders ]  The profit /income generated is the only long term possible way._**
 Now I try to answer you second part of the comment:- Applying Tensor flow and ML is needed very long steps and so many hours of research work to make them sit in correct position. For example  if you try with one method of deployment of Tensor flow and ML it needs so many sub- methods which need to be answered such as whether your data set is correct or not.? whether your  network training algo is correct or not.? or even whether you application language is correct or not.? Why correct application language must be taken into account.? Because if you we use normal Python package we can not expect more accuracy in retrieving the Tick data with respect to time factor. So in this aspect we must choose correct app language such as RUST or GO or Nim language than Python to get the LAtency factor say if we want the  time series without noise factor  issue and with more accuracy than the Normal binary tick duration of say 2 seconds per tick we may have to obtain that ticks in Neural network platform within milli second or micro second. For this we use GAN network to obtain the graphical output for training. 
The another flaw you faced in your tensor flow experiment is the **SPEED and DELAY in graphical output** 
That is due to non implementation of latest development in Tensor flow environment
Now in recent days the old Tensor flow method replaced by **Pytorch, Numpy** and even still more recent development with language like RUST or Nim [ instead of Python ] the speed is achieved and the issue regarding the dark request 0 response is totally  eliminated. 
How it is achieved .?
![1_5plivna5fiqec8-kz260kq](https://user-images.githubusercontent.com/28973275/53474332-cec27f00-3a92-11e9-98ae-85e0e8e6381d.gif)

1]They use modern **STACKED Neural Network** instead of conventional Neural network 
2] They now a days use Caps network ( Capsule Network ) in the place of CNN.
3]Wavelet transform using advanced mathematical formula  thereby **Lightening network connection without flaw is achieved to get back the data of binary TICK at very High Frequency rate** shown as below. That too without DASK delayed parallel Graph computation at continued manner..
The process is as follows: the data is transformed using Wavelet transform, then the remove coefficients that more than a full standard deviation away (out of all the coefficients), and inverse transform the new coefficients to get the denoised data.
![1_bwc7nnxrypte9tgaga8vog](https://user-images.githubusercontent.com/28973275/53473928-c6b60f80-3a91-11e9-8a50-9721b3e3fb8b.png)


Here is an example of how wavelet transform denoises time series data:
![1_f9nx1dkkezgky8n-v5asqw](https://user-images.githubusercontent.com/28973275/53473952-cf0e4a80-3a91-11e9-9724-79d9b8c3d127.gif)



As you can see, the random noise that is present in the initial signal is absent in the denoised versions. This is exactly what we are looking to do with our Binary TICK data
C**onclusion : It is very hard to implement most advanced latest AI based bot in short period. We need intensive research on this subject and we need dedicated time for this to happen and sleepless effort  with constant research mind can only achieve success in real and 100% working Binary trade Bots.**

**Comment by binary-ex-machina** [comment]
@amebamcare this is very nice stuff thanks for sharing. I'll take a look on it.

I agree that you need to keep researching to improve your trade strategies and style, but you can make money with binary options without using all this. I know a lot of traders that just look at the graph and trade (sometimes without any indicator) and make money everyday.

Using all this stuff you sent is for never need to open a graph or study about trading at all. You will have a computer running 24/7 making you rich, without you need much knowledge about trading. If only 10% of the trader make money, 0.01% of them are binary options traders. From that 0.01%, only 0.00000000000001% will be able to have a system like that. I'm sure this doesn't exist already, otherwise the brokers would have one something about or will not exist anymore.

**Comment by binarydream1** [comment]
I am quite surprised knowing the fact that only 0.01% binary options traders are making money. It is mean that a lot of us who learn binary option trading are wasting time and will eventually lose our money.

**Comment by binary-ex-machina** [comment]
@binarydream1 read it again. From all the traders in the world, 0.01% of them are trading binary options. I think it will be around 10-15% of the people who trade binary options make money on long term.

**Comment by amebamcare** [comment]
@binary-ex-machina  One clarification needed :-  Can we integrate your XML based block based binary options tradebot with Trading view .? Some people use this Trading view services via service known as **Auto view** google extension. If we are able to use this with your XML based block tradebot,  we can able to  automate this via cloud server. Also I heard about  even some people partially implemented it using of Machine language / A.I in **Trading view**.  So my question is : Can we integrate the XML based block bots with **Trading view**  as indicator for binary tick.? if yes how  to do this.? Thanks in advance

**Comment by binary-ex-machina** [comment]
@amebamcare you can't modify trading view inside binary.com or interact with them besides the web browser.

**Comment by aaron-binary** [comment]
We're closing this issue due to inactivity, you're welcome to open it again if you still haven't got your answer.

Regards,
Aaron

**Comment by Simon719** [comment]
Hi. But what about digit match/differs, even/odd strategy?
